docs/source/advanced/mixed_precision.rst

```diff
@@ -5,49 +5,154 @@
+
+Higher precision, such as the 64-bit floating-point, can be used for highly sensitive use-cases.
+
+Following are the precisions available in Lightning along with their supported Accelerator:
+
+.. list-table:: Precision with Accelerators
+   :widths: 20 20 20 20 20
+   :header-rows: 1
+
+   * - Precision
+     - CPU
+     - GPU
+     - TPU
+     - IPU
+   * - 16
+     - No
+     - Yes
+     - No
+     - Yes
+   * - BFloat16
+     - Yes
+     - Yes
+     - Yes
+     - No
+   * - 32
+     - Yes
+     - Yes
+     - Yes
+     - Yes
+   * - 64
+     - Yes
+     - Yes
+     - No
+     - No
+
+
+***************
+Mixed Precision
+***************
+
+PyTorch, like most deep learning frameworks, trains on 32-bit floating-point (FP32) arithmetic by default. However, many deep learning models do not require this to reach complete accuracy. By conducting
+operations in half-precision format while keeping minimum information in single-precision to maintain as much information as possible in crucial areas of the network, mixed precision training delivers
+significant computational speedup. Switching to mixed precision has resulted in considerable training speedups since the introduction of Tensor Cores in the Volta and Turing architectures. It combines
+FP32 and lower-bit floating-points (such as FP16) to reduce memory footprint and increase performance during model training and evaluation. It accomplishes this by recognizing the steps that require
+complete accuracy and employing a 32-bit floating-point for those steps only, while using a 16-bit floating-point for the rest. When compared to complete precision training, mixed precision training
+delivers all of these benefits while ensuring that no task-specific accuracy is lost. [`2 <https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html>`_].
 
 .. note::
 
-    In some cases it is important to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    In some cases, it is essential to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    For example, when running scatter operations during the forward (such as torchpoint3d), computation must remain in FP32.
+
+.. warning::
+
+    Do not cast anything to other dtypes manually using ``torch.autocast`` when using native precision because
+    this can bring instability.
+
+    .. code-block:: python
+
+        class LitModel(LightningModule):
+            def training_step(self, batch, batch_idx):
+                outs = self(batch)
+                target = torch.zeros_like(outs)
+                with torch.autocast(device_type=self.device.type, dtype=torch.float):
+                    # casting to float32
+                    outs = outs.softmax(dim=-1)
+
+                # here outs is of type float32 and target is of type float16
+                loss = F.mse_loss(outs, target)
```

Comment:
have you checked this? From what I thought this would not be true, as autocast only casts the inputs of the operations and casts the outputs back to the original dtype

Reply:
I checked this example.. if that would have been the case, wouldn't it have caused issues when calculating loss/metric with a target variable that comes from an input batch?

Reply:
i don't think so. From my understanding autocast behaves like this:

The contextmanager itself doesn't do much upon the entrance, but for every single torch operation all inputs are casted to the specified dtype (if supported by the operation) then the op is computed and all outputs are casted back to the original dtype. I don't see how this could cause issues and I think this is also the only way to have single layers (like batchnorm) excluded.

Reply:
updated as discussed